src/libraries/System.Private.CoreLib/src/System/Reflection/MethodInvoker.cs

```diff
@@ -2,78 +2,364 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Diagnostics;
+using static System.Reflection.InvokerEmitUtil;
 using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime;
+using static System.Reflection.MethodBase;
 
 namespace System.Reflection
 {
-    internal sealed partial class MethodInvoker
+    public sealed partial class MethodInvoker
     {
-        private readonly MethodBase _method;
+        internal InvokeFunc_ObjSpanArgs? _invokeFunc_ObjSpanArgs;
```

Comment:
Since MethodInvoker\ConstructorInvoker are public but MethodBaseInvoker is internal, that is not possible (having an internal base class on a public class). The original API proposal has a public base class that was to be shared, but that was rejected as not necessary during API review.